dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -167,23 +165,40 @@ function ContextSelector() {
               </select>
             </div>
             <div className="kubeapps-create-new-ns">
-              <CdsModal hidden={!newNSModalIsOpen} closable={true} onCloseChange={closeNewNSModal}>
-                <CdsModalHeader>Create a New Namespace</CdsModalHeader>
-                {error && <Alert theme="danger">An error occurred: {error.error.message}</Alert>}
-                <form onSubmit={createNewNS}>
-                  <CdsModalContent>
-                    <CdsFormGroup>
-                      <CdsInput>
-                        <label>Name:</label>
-                        <input type="text" required={true} onChange={onChangeNewNS} />
-                      </CdsInput>
-                    </CdsFormGroup>
-                  </CdsModalContent>
-                  <CdsModalActions>
-                    <CdsButton type="submit">Submit</CdsButton>
-                  </CdsModalActions>
-                </form>
-              </CdsModal>
+              <Modal
+                showModal={newNSModalIsOpen}
+                onModalClose={closeNewNSModal}
+                title="Create a New Namespace"
+              >
+                <div className="newns-modal">
+                  {error && <Alert theme="danger">An error occurred: {error.error.message}</Alert>}
+                  <form onSubmit={createNewNS}>
+                    <div className="clr-form-control">
+                      <label htmlFor="namespace-name" className="clr-control-label">
+                        Name
+                      </label>
+                      <div className="clr-control-container">
+                        <div className="clr-input-wrapper">
+                          <input
+                            type="text"
+                            className="clr-input"
+                            placeholder="my-namespace"
+                            onChange={onChangeNewNS}
+                            required={true}
+                            pattern="[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+                            title="Use lower case alphanumeric characters, '-' or '.'"
```

Comment:
I've added this validation (as in other forms), but let me know if I'm wrong here.

Reply:
I would not add the pattern validation, we may miss valid cases. I'd let the K8s API to do the validation for us.